app/helpers/indoor_routing_helper.rb

```diff
@@ -0,0 +1,53 @@
+module IndoorRoutingHelper
+  def self.update_nearest_node_result(door_id, dist, building, result)
+    result[:door] = door_id
+    result[:distance] = dist
+    result[:building] = building
+  end
+
+  def self.find_nearest_node(latlng, buildings, level, result)
+    buildings.each do |b|
+      graph = IndoorGraph.indoor_graphs[b]
+      IndoorGraph.nodes[b].each do |door_id|
+        next unless graph[door_id]["floor"] == level
+
+        dist = RoutingHelper.distance(graph[door_id]["latlng"], latlng)
+        update_nearest_node_result(door_id, dist, b, result) if dist < result[:distance]
+      end
+    end
+  end
+
+  def self.closest_node(latlng, buildings, max_dist, level)
+    result = { door: nil, distance: Float::MAX, building: nil }
+    find_nearest_node(latlng, buildings, level, result)
```

Comment:
Since this `result` in a lot of places that mutate it, maybe it would help if it was a small data class? Or is that not the Ruby way? (I don't know)